src/dataService/pocketSavesService.ts

```diff
@@ -0,0 +1,155 @@
+import { Knex } from 'knex';
+import { IContext } from '../server/context';
+import { PocketSaveStatus } from '../types';
+
+export type RawListResult = {
+  api_id: string;
+  api_id_updated: number;
+  favorite: number;
+  given_url: string;
+  item_id: number;
+  resolved_id: number;
+  status: number;
+  time_added: Date;
+  time_favorited: Date;
+  time_read: Date;
+  time_updated: Date;
+  title: string;
+  user_id: number;
+};
+
+export type ListResult = {
+  api_id: string;
+  api_id_updated: number;
+  favorite: number;
+  given_url: string;
+  item_id: number;
+  resolved_id: number;
+  status: keyof typeof PocketSaveStatus;
+  time_added: Date;
+  time_favorited: Date;
+  time_read: Date;
+  time_updated: Date;
+  title: string;
+  user_id: number;
+};
+
+/***
+ * class that handles the read and write from `readitla-temp.list` table
+ * note: for mutations, please pass the writeClient, otherwise there will be replication lags.
+ */
+export class PocketSaveDataService {
+  private db: Knex;
+  private readonly apiId: string;
+  private readonly userId: string;
+  private static statusMap = {
+    [PocketSaveStatus.UNREAD]: 'UNREAD',
+    [PocketSaveStatus.ARCHIVED]: 'ARCHIVED',
+    [PocketSaveStatus.DELETED]: 'DELETED',
+    [PocketSaveStatus.HIDDEN]: 'HIDDEN',
+  };
+
+  constructor(context: Pick<IContext, 'apiId' | 'dbClient' | 'userId'>) {
+    this.apiId = context.apiId;
+    this.db = context.dbClient;
+    this.userId = context.userId;
+  }
+
+  public static convertListResult(listResult: null): null;
+  public static convertListResult(listResult: RawListResult): ListResult;
+  public static convertListResult(listResult: RawListResult[]): ListResult[];
+  /**
+   * Convert the `status` field in the list table to the expected
+   * GraphQL ENUM string
+   * @param listResult
+   */
+  public static convertListResult(
+    listResult: RawListResult | RawListResult[] | null
+  ): ListResult | ListResult[] | null {
+    if (listResult == null) {
```

Comment:
I would probably go for `listResult === null` if you're interested in null in particular, or `!listResult` if you're interested in any falsy value.

Reply:
we use this pattern when the response could be null or undefined, but we don't care which. I like that it's a little more specific than just falsy values, which can be a little unintuitive in js/ts (e.g. [] is truthy, {} is truthy)

Reply:
Trying the more explicit `x === undefined || x === null`